src/common/legend/LegendDirective.js

```diff
@@ -116,7 +116,23 @@
               } else {
                 domain = server.url;
               }
-
+              // Proxy the domain if appropriate
+              var getUseProxyParam = function(server) {
+                return goog.isDefAndNotNull(server.use_proxy) ? server.use_proxy : false;
+              };
+              // proxy a url if the souce parameters indicate to do so
+              var useProxyUrlParam = function(use_proxy, url) {
+                if (goog.isDefAndNotNull(use_proxy) && use_proxy === true) {
+                  url = decodeURIComponent(url);
+                  if (url.indexOf(configService.configuration.proxy) < 0) {
+                    url = configService.configuration.proxy + encodeURIComponent(url);
+                  } else {
+                    url = encodeURIComponent(url);
```

Comment:
`encodeURIComponent(url)` makes sense for the `url=` value (where it encodes the '/', like Python's `quote_plus`), but does it make sense here in the else?

Reply:
It's already encoded when it gets there. I decode it first, then check if it contains the proxy already (just to be safe). If it doesn't, I add in the proxy and reecode it, otherwise just reencode it.